packages/strapi-plugin-users-permissions/services/Providers.js

```diff
@@ -45,7 +45,7 @@ const connect = (provider, query) => {
 
       try {
         const users = await strapi.query('user', 'users-permissions').find({
-          email: profile.email,
+          email: profile.email.toLowerCase(),
```

Comment:
this will break for existing users. you will need to run a or query for the lowercased and non lowercase email

Reply:
Couple of questions

1. Isn't there some sort of migration script that can lowercase all emails in the database when you upgrade? 

2. Does strapi truly not support non case sensitive database searches? Surely there is an option to pass in that equates to SELECT * FROM usertable WHERE LOWER(email) = LOWER(value)

Reply:
If there is no option for case insensitive search, the only real choice is to _always_ store lowercase values and _always_ compare with lowercased query strings. Which is fine, considering the following: 

1. There should be some method to "migrate" existing databases to convert to all lowercase usernames & emails when upgrading to the next release
2. For sake of testing and thoroughness, storing lowercase usernames & emails should be handled at the controller level.
3. Also for sake of testing and thoroughness, finding user accounts by lowercase usernames & emails should be handled at the controller level. 

The above will lead to a cleaner, more testable codebase with less opportunity for missed checks and inconsistencies in handling username & email fields.

Reply:
@bglidwell it would require a custom query to be built, the default `strapi.query().find()` doesn't do lookups like that.

Reply:
@alexandrebodin I don't have a good solution to this, MySQL is case insensitive (and I think PG and SQLite are also by default) and Mongo I believe is the problem.

We could use a `email ILIKE profile.email` if there is SQL that isn't but I'm not sure on mongo. If the rest of Strapi is lowering the emails it is probably best the providers are too and ask the users to migrate.

Reply:
(Alternative on SQL would be a `LOWER LIKE`  such as what @bglidwell suggested, but the problem would still be with MongoDB)

Reply:
We could do a `_or` query to check for this value or the lowercased one and save it to lowercase from now on. this will keep the code backwards compat. I am actually not sure that we save the admin emails in lowercased tbh... I think the transform is only done in the frontend 🤔 

e.g

```
strapi.query().find({ _where: { _or: [{email: profile.email.lowerCase}, { email: profile.email }]}}
```

Reply:
Think this is the first time I've seen the _where being used in the code :laughing: I'll just the PR to that then, but we should adjust them for the databases that treat all normal searches as case sensitive. 

Most email specs I've seen do not treat email addresses as case sensitive, so from a standards perspective it shouldn't matter if we forcefully save them to lowercase.

Reply:
Just my $0.02, but essentially everywhere in the codebase that a username or email is CREATED, UPDATED, or DELETED strapi will need to store lowercase, and everywhere a username or email is READ/QUERIED strapi will need to match with the lowercase of the query string (if included) OR the original string. Then this needs to be done for both user info and admin info, and perhaps even on frontend of admin panel. It seems like this really justifies having it's own query/controller, no? There has to be several dozen spots where this toLowerCase() massaging will have to take place. If a separate "finduser" for example were done, it would be simpler (or at least cleaner) to put all this logic in one place and update the codebase with references to this instead. I don't propose to know the architecture well enough to propose HOW that might be done, but user accounts and authentication are important enough to get right the first time.

Reply:
I don't disagree with you @bglidwell as what you are saying makes sense to me.

@alexandrebodin @Convly @petersg83 would you rather me close this and you guys can handle it at a higher level? I was trying for a simple fix but the convo as gone beyond that :laughing:

Reply:
This can wait until a major release so we can make a clean breaking change and tell people to run a db migration to put all the emails to lowercase IMO

Reply:
Sounds good to me, you want me to leave this open or just make a note on the parent issue and close this PR?

Reply:
We can leave this open as a ref but we might want to add some labels for PR that are in standby (already some waiting to add a breaking change)